### ISSUE TITLE
refactor `Addon.update_nominated_status` into `Addon.update_status`

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#19773]
follow-up for  #19728 - `update_nominated_status` was added to change the add-on status to `STATUS_NOMINATED` if there are any listed versions that are `STATUS_AWAITING_REVIEW` and the addon status is `STATUS_NULL`, but we should be able to **carefully** fold this functionality into `update_status` directly...  Also see thread https://github.com/mozilla/addons-server/pull/19728#discussion_r984674335

**Comment by diox** [comment]
Note: ideally, we should use that occasion to clean up any code that directly assigns a `status` to an `Addon` instance outside of the methods in `Addon` themselves.

**Comment by eviljeff** [comment]
It's worth considering if we want to delegate_all_ status setting to `update_status` - i.e. would it just apply to `STATUS_NOMINATED` upgrades, or are we comfortable with the function upgrading an add-ons status to `STATUS_APPROVED` without any explicit user action too?   I guess these days the worst that can happen would be an add-on that is "approved" with no signed versions.

**Comment by diox** [comment]
The way I see it, in an ideal world, `status` should just be a denormalized field that makes our lives easier and can always be safely recomputed at all times. We're not 100% there because of force disable/enable actions, but apart from that, I think we should be pretty close.

With that philosophy in mind I think it's ok to have a single function that's responsible for all `status` changes. We could split it into two ore more, but I am not sure it would buy us much, I feel that would just increase complexity.

**Comment by eviljeff** [comment]
I have a working-ish* patch for this at https://github.com/mozilla/addons-server/compare/master...eviljeff:olympia:19773-refactor-addon-update-nominated-status but fixing the tests made me realise that it's not enough to have `update_status` update the status automatically - we really need to entirely stop manually changing the add-on status outside of the `Addon` model to avoid unexpected status changes in real operation, and to have accurate tests that reflect reality on prod.

The other problem we have is the requirement to have complete metadata to become `STATUS_NOMINATED` - it's not clear if it should be hard requirement (i.e. to be `STATUS_NOMINATED` the add-on must both have a `File` that is `STATUS_AWAITING_REVIEW`, _and_ have complete metadata, in all cases; or only apply the metadata restriction when the existing status is `STATUS_NULL`)

Having already spent the best part of two days on this I'm putting it down again.

* just a [few remaining](https://app.circleci.com/pipelines/github/eviljeff/olympia?branch=19773-refactor-addon-update-nominated-status) test fails, mainly around unexpected activity logs

**Comment by eviljeff** [comment]
QA: this changes how Addon status is determined so there could be impacts anywhere, but in particular check:
- A new add-on submitted with a listed version gains Nominated status
  - repeat for addon submission api; signing api; devhub
- the unreject versions action (with listed versions) in reviewer tools correctly sets the addon status back to Nominated if there are no other approved versions.
- request review and cancel review actions in devhub work as expected.

**Comment by ioanarusiczki** [comment]
@eviljeff 

- I made submissions of listed addons and I noticed in the API responses a "status": "nominated", which is available until the first version is approved.  The "nominated" status is only available in the APIs (the addon detail API exposes it or the response of the new submission API). A first submitted version is still with "Awaiting review" in dev hub or on it's review page so nothing seems to be different than before.

- when there is no other approved version, if I unreject the status will go back to "status": "nominated".

- I requested admin add-on-review and admin content-review  -> these are actions that will be registered by the Activity Log. Then I could cancel them using the button Clear Admin Code Review Flag (this is not registered by the Activity Log but it's the same on -stage, means it's been like that)

I'm going to mark this verified after I'll look on AMO stage over dev hub and rev tools pages at release testing.
